### PR TITLE
Seal of Approval additions

### PR DESCRIPTION
**Pull request by kbodwin**
PR'ing to add the Seal_of_Approval.md document that keeps a list of approved packages, as well as proposing the first three pilot packages for approval.

(Also updating R Build Ignore and README accordingly.)

### CODE REVIEW

File: Seal_of_Approval.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+This is a list of packages in the `data.table` community.
+
+Further detail about these packages and their relationship to `data.table` can be found at [The Raft blog](https://rdatatable-community.github.io/The-Raft/#category=seal%20of%20approval).
+
+To add your package to this list, please [submit a Pull Request to The Raft](https://github.com/rdatatable-community/The-Raft/), making sure to follow the templated instructions.
+
+## Extension packages: Add to the internal functionality of `data.table`
+
+- [nc](https://github.com/tdhock/nc): Named capture regular expressions for text parsing and data reshaping.
```
Comment (Anirban166) [inline Seal_of_Approval.md:11]: Maybe 'Named capture groups in regular expressions ... '?

Comment (tdhock) [reply Seal_of_Approval.md:11]: this was copied from https://github.com/tdhock/nc

**Comment by tdhock** [comment]
looks good to me, thanks Kelly.
Since there are some significant changes to the README, and I am involved with this PR, I will not merge this.
Can another committer please review and merge? @jangorecki @ben-schwen

File: Seal_of_Approval.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+## Application packages: Use `data.table` to accomplish a particular task or analysis.
+
+## Bridge package: Translate data.table syntax to different syntax or provides helper functions for transitioning between `data.table` and another object type.
+
+- [tidyfast](https://github.com/TysonStanley/tidyfast): Fast and efficient alternatives to tidyr functions built on data.table
+
+- [dtplyr](https://github.com/tidyverse/dtplyr): A `data.table` backend for `dplyr`
+
+## Partner package: Not necessarily directly connected to `data.table`, but deliberately follows the [core philosophies of `data.table`](https://github.com/Rdatatable/data.table/blob/master/GOVERNANCE.md).
```
Comment (jangorecki) [inline Seal_of_Approval.md:21]: This link is wrong, no?

**Comment by jangorecki** [comment]
Looks good, left one minor comment

File: Seal_of_Approval.md (hunk 2 above)
Comment (TysonStanley) [reply Seal_of_Approval.md:21]: I believe this is referring to the six that we list in the governance.

- Time & memory efficiency
- Concise syntax (minimal redundancy in code)
- No external Imports/LinkingTo/Depends dependencies (external meaning those not maintained by the project)
- Few (if any) Suggests/Enhances dependencies
- Stable code base (strong preference for user-friendly back-compatibility with data.table itself and with old versions of R)
- Comprehensive and accessible documentation and run-time signals (errors, warnings)

File: Seal_of_Approval.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+## Extension packages: Add to the internal functionality of `data.table`
+
+- [nc](https://github.com/tdhock/nc): Named capture regular expressions for text parsing and data reshaping.
+
+## Application packages: Use `data.table` to accomplish a particular task or analysis.
+
+## Bridge package: Translate data.table syntax to different syntax or provides helper functions for transitioning between `data.table` and another object type.
+
+- [tidyfast](https://github.com/TysonStanley/tidyfast): Fast and efficient alternatives to tidyr functions built on data.table
```
[the remark this answers is not in the document]
Comment (TysonStanley) [reply Seal_of_Approval.md:17]: It is designed differently than `dtplyr` so I'm fine with how it is currently stated.

**Review by TysonStanley — APPROVED** [review]
LGTM

File: Seal_of_Approval.md (hunk 2 above)
Comment (jangorecki) [reply Seal_of_Approval.md:21]: can we link that section of the file?

Comment (tdhock) [reply Seal_of_Approval.md]: I did

**Comment by tdhock** [comment]
great thanks!

**Review by jangorecki — COMMENTED** [review]
Lgtm

**Pull request merged**